packages/server/src/adapters/node-http/incomingMessageToRequest.ts

```diff
@@ -0,0 +1,85 @@
+import type * as http from 'http';
+import { TRPCError } from '../../@trpc/server';
+
+/**
+ * Convert an incoming message to a body stream with a max size
+ */
+function incomingMessageToBodyStream(
+  req: http.IncomingMessage,
+  opts: { maxBodySize: number | null },
+) {
+  type Value = Buffer | Uint8Array | string | null;
+  let size = 0;
+  const maxBodySize = opts.maxBodySize;
+
+  let controller: ReadableStreamDefaultController<Value> =
+    null as unknown as ReadableStreamDefaultController<Value>;
+  const stream = new ReadableStream<Value>({
+    start(c) {
+      controller = c;
+    },
+    async pull(c) {
+      const chunk: Value = req.read();
+
+      if (chunk) {
+        size += chunk.length;
+      }
+      if (maxBodySize !== null && size > maxBodySize) {
+        controller.error(
+          new TRPCError({
+            code: 'PAYLOAD_TOO_LARGE',
+          }),
+        );
+        return;
+      }
+      if (chunk === null) {
+        c.close();
+        return;
+      }
+      controller.enqueue(chunk);
+    },
+    cancel() {
+      req.destroy();
+    },
```

Comment:
Considering dropping maxBodySize in order to simplify this, it can be replaced by just using the IncomingMessage as the body (as it is a stream)